pkg/util/aad/aad.go

```diff
@@ -6,29 +6,92 @@ package aad
 import (
 	"context"
 	"net/http"
+	"strings"
 	"time"
 
+	azgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
 	"github.com/Azure/go-autorest/autorest/azure/auth"
 	"github.com/form3tech-oss/jwt-go"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/graphrbac"
 	"github.com/Azure/ARO-RP/pkg/util/refreshable"
 )
 
+type Manager interface {
+	GetToken(ctx context.Context, resource string) (*adal.ServicePrincipalToken, error)
+	GetServicePrincipalID(ctx context.Context) (string, error)
+}
+
+type manager struct {
+	log          *logrus.Entry
+	azEnv        *azure.Environment
+	tenantID     string
+	clientID     string
+	clientSecret string
+}
+
+// NewManager returns aad manager for AAD operations.
+func NewManager(log *logrus.Entry, azEnv *azure.Environment, tenantID, clientID, clientSecret string) Manager {
+	// manager should be kept abstract enough so we could call it from cluster
+	// or RP context.
+	return &manager{
+		log:          log,
+		azEnv:        azEnv,
+		tenantID:     tenantID,
+		clientID:     clientID,
+		clientSecret: clientSecret,
+	}
+}
+
+func (m *manager) GetServicePrincipalID(ctx context.Context) (string, error) {
```

Comment:
If we are calling this a lot, should we actually be storing the spID in the internal model?

Reply:
I think this makes sense. As separate PR would be good.